lib/auth0/mixins/validation.rb

```diff
@@ -85,15 +86,140 @@ def decode_and_validate_signature(id_token, header)
             raise Auth0::InvalidIdToken, 'Invalid ID token signature'
           rescue JWT::IncorrectAlgorithm
             alg = header['alg']
-            raise Auth0::InvalidIdToken, "Signature algorithm of \"#{alg}\" is not supported. Expected the ID token to be signed with \"#{algorithm.name}\""
+            raise Auth0::InvalidIdToken, "Signature algorithm of \"#{alg}\" is not supported. Expected the ID token"\
+                                         " to be signed with \"#{algorithm.name}\""
           rescue JWT::DecodeError
             raise Auth0::InvalidIdToken, "Could not find a public key for Key ID (kid) \"#{kid}\""
           end
         end
-        # rubocop:enable Metrics/MethodLength, Metrics/AbcSize, Metrics/CyclomaticComplexity
 
-        def validate_claims(decoded_id_token); end # TODO: Implement (in the next PR)
+        # rubocop:disable Metrics/PerceivedComplexity
+        def validate_claims(claims)
+          leeway = @context[:leeway]
+          nonce = @context[:nonce]
+          issuer = @context[:issuer]
+          audience = @context[:audience]
+          max_age = @context[:max_age]
+
+          raise Auth0::InvalidParameter, 'Must supply a valid leeway' unless leeway.is_a?(Integer) && leeway >= 0
+          raise Auth0::InvalidParameter, 'Must supply a valid nonce' unless nonce.nil? || !nonce.to_s.empty?
+          raise Auth0::InvalidParameter, 'Must supply a valid issuer' unless issuer.nil? || !issuer.to_s.empty?
+          raise Auth0::InvalidParameter, 'Must supply a valid audience' unless audience.nil? || !audience.to_s.empty?
+
+          unless max_age.nil? || (max_age.is_a?(Integer) && max_age >= 0)
```

Comment:
this one looks similar to the first check a few lines above for leeway. What's the reason for using a different style here? remember leeway could also be set to `nil` by the dev

Reply:
It's because a single line would get too long.

Reply:
The leeway is one param that is not expected to be `nil`, since it has a default value. Of course the validation will fail if it's `nil`, since it checks that the value is an integer, but is not a parameter in which `nil` is a valid option. If the dev does not want a leeway, the way to do that is to set it to `0`.